br/pkg/lightning/mydump/loader.go

```diff
@@ -422,6 +423,10 @@ func (s *mdLoaderSetup) route() error {
 			remainingSchemas = append(remainingSchemas, info)
 		}
 	}
+	// if there are no tables in the exported data, we still need route s.dbSchemas(remainingSchemas)
+	if err := run(remainingSchemas); err != nil {
```

Comment:
`knownDBNames[web].count` is larger than 0 because its count equals `1 + COUNT(table_schema_file) + COUNT(table_data_file) + COUNT(table_view_schema)`, where `1` is the database schema file.

In the situation where `COUNT(table_data_file)==0`, `knownDBNames[web].count -= COUNT(table_schema_file)` and gets `1` because the database schema file is not routed. Maybe u can try routing `dbSchemas` (just my hypothesis...)

`remainingSchemas` is the result of routing. Processing it again does lead to the problem @gozssky mentioned.

FYI: https://github.com/pingcap/tidb/blob/b2cef4e8783a06a416348cc7a85b321ae0486fda/br/pkg/lightning/mydump/loader.go#L364-L369

Reply:
thanks I'll look in detail